test/compilation_test.rb

```diff
@@ -33,9 +35,52 @@ def copy_example_file(filename)
   end
 
   test "asset pipeline should serve foo.js" do
-    expected_output = fixture("foo.out.js")
+    expected_output = fixture("application.out.js")
+
+    get "/assets/application.js"
+    assert_response :success
+    assert_equal expected_output, @response.body.strip
+  end
+
+  test "asset pipeline should regenerate application.js when node_modules changes" do
+    Dummy::Application.config.browserify_rails.evaluate_node_modules = false
+    expected_output = fixture("application.out.js")
+
+    get "/assets/application.js"
+    assert_response :success
+    assert_equal expected_output, @response.body.strip
+
+    # Ensure that Sprockets can detect the change to the file modification time
+    sleep 1
+
+    File.open(File.join(Rails.root, "node_modules/node-test-package/index.js"), "w+") do |f|
+      f.puts 'module.exports = console.log("goodbye friend");'
+    end
+
+    expected_output = fixture("application.out.js")
 
-    get "/assets/foo.js"
+    get "/assets/application.js"
+    assert_response :success
+    assert_equal expected_output, @response.body.strip
+  end
+
+  test "asset pipeline should not regenerate application.js when node_modules changes" do
```

Comment:
I'm somewhat new to browserify and browserify-rails, so I may be missing something, but: is this test name accurate?  From what I can tell, it looks like application.js **is** regenerated in this test.